### PR TITLE
bugfix:add clusterfile to all image whatever build type is

### LINKED ISSUE #107: build image local mode not set Clusterfile to image metadata
#### What happened:

registry.cn-qingdao.aliyuncs.com/sealer-io/cloudrootfs:v1.19.9-alpha.3 

sealer build -t  registry.cn-qingdao.aliyuncs.com/sealer-io/cloudrootfs:v1.19.9-alpha.3  . 

cat /var/lib/sealer/metadata/9ce2938527c2aff4b95e78fd961b7a19.yaml

```
kind: Image
metadata:
  creationTimestamp: null
  name: registry.cn-hongkong.aliyuncs.com/sealyun/kubernetes-amd64:v1.19.9
spec:
  hash: ""
  id: 9ce2938527c2aff4b95e78fd961b7a19
  layers:
  - hash: 245d2093fabc329a6654751f63999d7fec433de59a4ed7ca244398e2c01582ba
    type: COPY
    value: . .
  mergedLayer: ""
status: {}
```

not clusterfile config

#### Environment:
- sealer version (use `sealer version`):
- Cloud provider or hardware configuration:
- OS (e.g: `cat /etc/os-release`):
- Kernel (e.g. `uname -a`):
- Install tools:
- Others:

### PR DESCRIPTION
**Pull request by kakaZhou719**
### Describe what this PR does / why we need it
support to add clusterfile to all image whatever build type is

### Does this pull request fix one issue?

Fixes  #107
### Describe how you did it

local build : find cluster file from kubefile, image matadata,rootfs .

cloud build : first find clusterfile from its applied, if not exist ,find it from local build steps.

if base image  is scratch, then read clusterfile from etc/Clusterfile
### Describe how to verify it
run local build: sealer build -f Kubefile -t {your image name } -b local .
run cloud build :  sealer build -f Kubefile -t {your image name } .
run build rootfs : from  scratch

### CODE REVIEW

**Review by fanux — APPROVED** [review]

**Comment by fanux** [comment]
LGTM need test

File: build/local_builder.go [hunk 1]
```diff
@@ -276,23 +277,11 @@ func (l *LocalBuilder) ApplyCluster() error {
 }
 
 func (l *LocalBuilder) UpdateImageMetadata() error {
+	l.setClusterFileToImage()
 	if err := l.squashBaseImageLayerIntoCurrentImage(); err != nil {
 		return err
 	}
-	// write image info to its metadata
 	filename := fmt.Sprintf("%s/%s%s", common.DefaultImageMetaRootDir, l.Image.Spec.ID, common.YamlSuffix)
```
Comment (justadogistaken) [inline build/local_builder.go:284]: "%s/%s.%s"

File: build/local_builder.go [hunk 2]
```diff
@@ -324,6 +313,65 @@ func (l *LocalBuilder) UpdateImageMetadata() error {
 	return nil
 }
 
+// setClusterFileToImage: set cluster file whatever build type is
+func (l *LocalBuilder) setClusterFileToImage() {
+	var clusterFileData string
+	if utils.IsFileExist(common.RawClusterfile) {
+		bytes, err := utils.ReadAll(common.RawClusterfile)
+		if err != nil {
+			logger.Warn("failed to set cluster file to image metadata,%s not exist", common.RawClusterfile)
```
Comment (justadogistaken) [inline build/local_builder.go:322]: do not ignore the err.

Comment (justadogistaken) [reply build/local_builder.go:322]: better return the err

File: build/local_builder.go [hunk 3]
```diff
@@ -324,6 +313,65 @@ func (l *LocalBuilder) UpdateImageMetadata() error {
...
+		clusterFileData = string(bytes)
+	} else {
+		clusterFileData = l.GetRawClusterFile()
+	}
+
+	l.addImageAnnotations(common.ImageAnnotationForClusterfile, clusterFileData)
+}
+
+// GetClusterFile from user build context or from base image
+func (l *LocalBuilder) GetRawClusterFile() string {
```
Comment (justadogistaken) [inline build/local_builder.go:333]: return err, let caller decide how to dispose

File: build/local_builder.go [hunk 4]
```diff
@@ -324,6 +313,65 @@ func (l *LocalBuilder) UpdateImageMetadata() error {
...
+	// find cluster file from base image
+	clusterFile := image.GetClusterFileFromImage(l.Image.Spec.Layers[0].Value)
+	if clusterFile != "" {
+		logger.Info("get cluster file from base image success!")
+		return clusterFile
+	}
+	return ""
+}
+
+func (l *LocalBuilder) getClusterFileFromContext() []byte {
```
Comment (justadogistaken) [inline build/local_builder.go:355]: err

**Pull request merged**